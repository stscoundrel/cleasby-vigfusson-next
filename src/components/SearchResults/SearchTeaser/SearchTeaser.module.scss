.result {
  max-width: 500px;
  margin-bottom: 30px;
}

.link {
  display: block;
  max-width: 160px;
  padding: 2px 0;
  margin: 4px 0;
}

.foundIn {
  font-size: smaller;
}