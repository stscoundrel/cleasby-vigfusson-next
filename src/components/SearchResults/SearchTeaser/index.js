import Link from 'next/link'
import styles from './SearchTeaser.module.scss'

export default function SearchTeaser({ data }) {
  const { slug, word, foundIn } = data

  return (
    <div className={styles.result}>
      <Link key={`link${slug}`} href={`/word/${slug}`}>
        <a className={styles.link}>{word.toLowerCase()}</a>
      </Link>
      <ul>
        { foundIn.map((searchResult, index) => (
          <li
            className={styles.foundIn}
            key={`${data.slug}-search-${index}`}
            dangerouslySetInnerHTML={ { __html: searchResult } }
          />
        )) }
      </ul>
    </div>
  )
}
