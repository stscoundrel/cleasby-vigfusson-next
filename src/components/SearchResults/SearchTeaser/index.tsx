import Link from 'next/link'
import { SearchResult } from 'lib/services/search';
import styles from './SearchTeaser.module.scss'

interface SearchTeaserProps {
  data: SearchResult
}

export default function SearchTeaser({ data }: SearchTeaserProps) {
  const { slug, word, foundIn } = data

  return (
    <div className={styles.result}>
      <Link key={`link${slug}`} href={`/word/${slug}`} className={styles.link} prefetch={false}>
        {word.toLowerCase()}
      </Link>
      <ul>
        { foundIn.map((searchResult, index) => (
          <li
            className={styles.foundIn}
            key={`${data.slug}-search-${index}`}
            dangerouslySetInnerHTML={ { __html: searchResult } }
          />
        )) }
      </ul>
    </div>
  );
}
