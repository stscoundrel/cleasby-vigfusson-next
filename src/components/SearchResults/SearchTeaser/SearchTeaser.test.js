import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import SearchTeaser from './index'

const word = {
  word: 'aðal-vellir',
  definitions: [
    'm. pl. = óðalvellir, Rm.',
  ],
  slug: 'adal-vellir',
  foundIn: [
    'In headword',
  ],
}

describe('Search teaser component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<SearchTeaser data={word} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<SearchTeaser data={word} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(<SearchTeaser data={word} />)
    const { root } = tree

    expect(root.findByType('a').children).toEqual([word.word])
  })
})
