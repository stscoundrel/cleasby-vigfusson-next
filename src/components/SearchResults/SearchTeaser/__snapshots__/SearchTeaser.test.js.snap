// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search teaser component Matches snapshot 1`] = `
<div
  className="result"
>
  <a
    className="link"
    href="/word/adal-vellir"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    a√∞al-vellir
  </a>
  <ul>
    <li
      className="foundIn"
      dangerouslySetInnerHTML={
        Object {
          "__html": "In headword",
        }
      }
    />
  </ul>
</div>
`;
